src/widget/scroll.rs

```diff
@@ -239,30 +292,115 @@ impl<T: Data, W: Widget<T>> Widget<T> for Scroll<T, W> {
+                BarHeldState::VerticalHeld(_) | BarHeldState::HorizontalHeld(_) => true,
+                _ => false,
+            },
+            _ => false,
+        } {
+            match event {
+                Event::MouseMoved(event) => {
+                    match self.scroll_bars.held {
+                        BarHeldState::VerticalHeld(offset) => {
+                            let scale_y = viewport.height() / self.child_size.height;
+                            let bounds = self.calc_vertical_bar_bounds(&viewport);
+                            let mouse_y = event.pos.y + self.scroll_offset.y;
+                            let delta = mouse_y - bounds.y0 - offset;
+                            self.scroll(Vec2::new(0f64, (delta / scale_y).ceil()), size);
+                        }
+                        BarHeldState::HorizontalHeld(offset) => {
+                            let scale_x = viewport.width() / self.child_size.width;
+                            let bounds = self.calc_horizontal_bar_bounds(&viewport);
+                            let mouse_x = event.pos.x + self.scroll_offset.x;
+                            let delta = mouse_x - bounds.x0 - offset;
+                            self.scroll(Vec2::new((delta / scale_x).ceil(), 0f64), size);
+                        }
+                        _ => (),
+                    }
+                    ctx.invalidate();
+                }
+                Event::MouseUp(_) => {
+                    self.scroll_bars.held = BarHeldState::None;
                 }
+                _ => (),
             }
-            Event::Timer(id) if *id == self.scroll_bars.timer_id => {
-                // Schedule scroll bars animation
-                ctx.request_anim_frame();
-                self.scroll_bars.timer_id = TimerToken::INVALID;
+        } else if match event {
+            Event::MouseMoved(event) | Event::MouseDown(event) => {
+                let mut transformed_event = event.clone();
+                transformed_event.pos += self.scroll_offset;
+                self.point_hits_vertical_bar(viewport, transformed_event.pos)
+                    || self.point_hits_horizontal_bar(viewport, transformed_event.pos)
+            }
+            _ => false,
+        } {
+            match event {
+                Event::MouseMoved(event) => {
+                    let mut transformed_event = event.clone();
+                    transformed_event.pos += self.scroll_offset;
+                    if self.point_hits_vertical_bar(viewport, transformed_event.pos) {
+                        self.scroll_bars.hovered = BarHoveredState::VerticalHovered;
+                    } else {
+                        self.scroll_bars.hovered = BarHoveredState::HorizontalHovered;
+                    }
+
+                    self.scroll_bars.opacity = 0.7;
+                    self.scroll_bars.timer_id = TimerToken::INVALID; // Cancel any fade out in progress
```

Comment:
is there a reason we can't use `reset_scrollbar_fade` here?

Reply:
Yes because `reset_scrollbar_fade` starts the timer for a fade out